log/event_print.go

```diff
@@ -91,9 +92,17 @@ func generateStepFinishedFooterLines(params StepFinishedParams) []string {
 	}
 
 	var lines []string
+
+	for _, stepError := range params.Errors {
+		lines = append(lines, colorstring.Red(stepError.Message))
+	}
+	if params.StatusReason != "" {
+		lines = append(lines, colorstring.Blue(params.StatusReason))
+	}
```

Comment:
I have to admit you tricked me with this clever solution. 😂

I set a step preparation failed error and was debugging who prints the error inside the step box because at first look it did not make sense. The step did not run and still someone printed the error there. And if I ran it with json output then there were only the step started and step finished events and nothing in between.

But then I pieced it together and realised that the step footer now prints the errors and status reason as part of the main step body. It is was tricky for me to understand because my mind thought that the step footer always starts with the step timing information. 

I do not have a better solution but just wanted highlight that this might be a little bit confusing the first time you encounter it.